File: 53-maximum-subarray/53-maximum-subarray.java
class Solution {
    public int maxSubArray(int[] nums) {
        int max = Integer.MIN_VALUE;
        int val = 0;
        for (int num : nums) {
            val += num;
            max = Math.max(max, val);
            if (val < 0) val = 0;
        }
        return max;
    }
}

File: 53-MaximumSubarray/53-MaximumSubarray.cpp
// Last updated: 2026. 7. 16. 오전 11:28:33
class Solution {
public:
    int maxSubArray(vector<int>& nums) {
        int sum = 0;
        int sumMax = INT_MIN;
        for (int num : nums) {
            sum += num;
            sumMax = max(sumMax, sum);
            if (sum < 0) {
                sum = 0;
            }
        }
        return sumMax;
    }
};